google/compute_beta_operation.go

```diff
@@ -0,0 +1,167 @@
+package google
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/resource"
+
+	computeBeta "google.golang.org/api/compute/v0.beta"
+)
+
+// OperationBetaWaitType is an enum specifying what type of operation
+// we're waiting on from the beta API.
+type ComputeBetaOperationWaitType byte
+
+const (
+	ComputeBetaOperationWaitInvalid ComputeBetaOperationWaitType = iota
+	ComputeBetaOperationWaitGlobal
+	ComputeBetaOperationWaitRegion
+	ComputeBetaOperationWaitZone
+)
+
+type ComputeBetaOperationWaiter struct {
+	Service *computeBeta.Service
+	Op      *computeBeta.Operation
+	Project string
+	Region  string
+	Type    ComputeBetaOperationWaitType
+	Zone    string
+}
+
+func (w *ComputeBetaOperationWaiter) RefreshFunc() resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		var op *computeBeta.Operation
+		var err error
+
+		switch w.Type {
+		case ComputeBetaOperationWaitGlobal:
+			op, err = w.Service.GlobalOperations.Get(
+				w.Project, w.Op.Name).Do()
+		case ComputeBetaOperationWaitRegion:
+			op, err = w.Service.RegionOperations.Get(
+				w.Project, w.Region, w.Op.Name).Do()
+		case ComputeBetaOperationWaitZone:
+			op, err = w.Service.ZoneOperations.Get(
+				w.Project, w.Zone, w.Op.Name).Do()
+		default:
```

Comment:
`ComputeBetaOperation` is a copy of the existing `ComputeOperation` where we sub `compute/v0beta`'s `Operation` type for `compute/v1`'s `Operation` type. These changes would also need to be applied to `compute_operation.go` as well as `compute_beta_operation.go`. and I don't think that kind of change makes sense in the context of this PR.

Improving Operations is something that I've been thinking about with #187, which would introduce a standard `Operation` type for all versioned compute resources (such as storing them all as `v1` Operations), and a refactoring of `compute_operation.go` that was pulled into #191.